# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1468259112 14400
#      Mon Jul 11 13:45:12 2016 -0400
# Node ID c3823cb0ea02eb0cd497df11f778dcc38c791fd9
# Parent  7f79c716876cb28e9bd736ea13c1afef3e96a0eb
eliminate unused class definition

* octave-gui.cc (octave_cli_thread): Delete unused class.

diff --git a/libgui/src/octave-gui.cc b/libgui/src/octave-gui.cc
--- a/libgui/src/octave-gui.cc
+++ b/libgui/src/octave-gui.cc
@@ -34,63 +34,28 @@ along with Octave; see the file COPYING.
 #if defined (HAVE_SYS_IOCTL_H)
 #  include <sys/ioctl.h>
 #endif
 
 #include "lo-utils.h"
 #include "oct-env.h"
 #include "oct-syscalls.h"
 
-#include "octave.h"
-
-#include "welcome-wizard.h"
-#include "resource-manager.h"
-#include "shortcut-manager.h"
-#include "main-window.h"
-#include "octave-gui.h"
-#include "thread-manager.h"
-
 #include "builtin-defun-decls.h"
-
 #if defined (HAVE_QT_GRAPHICS)
 #  include "__init_qt__.h"
 #endif
-
-// Allow the Octave interpreter to start as in CLI mode with a
-// QApplication context so that it can use Qt for things like plotting
-// and UI widgets.
-
-class octave_cli_thread : public QThread
-{
-public:
-
-  octave_cli_thread (int argc, char **argv)
-    : m_argc (argc), m_argv (argv), m_result (0) { }
-
-  int result (void) const { return m_result; }
-
-protected:
+#include "octave.h"
 
-  void run (void)
-  {
-    octave_thread_manager::unblock_interrupt_signal ();
-
-    octave_initialize_interpreter (m_argc, m_argv, 0);
-
-    m_result = octave_execute_interpreter ();
-
-    QApplication::exit (m_result);
-  }
-
-private:
-
-  int m_argc;
-  char** m_argv;
-  int m_result;
-};
+#include "main-window.h"
+#include "octave-gui.h"
+#include "resource-manager.h"
+#include "shortcut-manager.h"
+#include "thread-manager.h"
+#include "welcome-wizard.h"
 
 // Disable all Qt messages by default.
 
 static void
 message_handler (QtMsgType, const char *)
 {
 }
 
